gene2phenotype_project/gene2phenotype_app/serializers.py

```diff
@@ -1099,16 +1096,41 @@ def create(self, validated_data):
                     raise serializers.ValidationError({"message": f"Data already submited to G2P '{lgd_obj.stable_id.stable_id}'"})
                 else:
                     raise serializers.ValidationError({"message": f"This is an old G2P record '{lgd_obj.stable_id.stable_id}'"})
+        
+        else:
+            raise serializers.ValidationError({"message" : "To publish a curated entry, locus, allelic requirement and disease are neccessary"})
+                                    
 
+    @transaction.atomic
+    def create(self, validated_data):
+        """
+            Create a new entry in the CurationData table.
+        
+            Parameters:
+                validated_data (dict): Validated data containing the JSON data to be stored.
+        
+            Returns:
+                CurationData: The newly created CurationData instance.
+            Future:
+                - update endpoint: updates the JSON data in existing session being curated
+                - publish endpoint: add the data to the G2P tables. entry will be live
+        """
+        json_data = validated_data.get("json_data")
+       
+        date_created = datetime.now()
+        date_reviewed = date_created
+        session_name = json_data.get('session_name')
         stable_id = G2PStableIDSerializer.create_stable_id()
-
-        # If curator did not input a session name, then the session name is the G2P stable ID
         if session_name == "":
-            session_name = stable_id.stable_id
+            session_name = stable_id
 
+         
+        user_email = self.context.get('user') # this needs to be looked at 
+        user_obj = User.objects.get(email=user_email)
+            
         new_curation_data = CurationData.objects.create(
             session_name=session_name,
-            json_data=data_json_input,
+            json_data=json_data,
             stable_id=stable_id,
```

Comment:
This is saving the instance "G2PStableID" as the session_name. We want to save the string, not the instance.

Reply:
Sorry, this comment is for the session_name, not the stable_id.